frontend/src/components/module-editor/SideBar.tsx

```diff
@@ -58,6 +80,40 @@ const Sidebar = (): React.ReactElement => {
   if (!context) return <></>;
   const { state, dispatch } = context;
 
+  const formatCourseRequest = (
+    newModule?: ModuleRequest,
+  ): [string, CourseRequest] => {
+    if (!courseData || error)
+      throw Error(
+        "Attempted to edit module when course does not contain modules",
+      );
+    let newModules = [];
+    // Copy other modules by reference due to the immutability of the data
+    if (newModule) {
+      // If module index isn't valid, append the new module
+      if (
+        Number(moduleIndex) >= 0 &&
+        Number(moduleIndex) < state.course.modules.length
+      )
+        newModules = state.course.modules.map((oldModule, index) =>
+          Number(moduleIndex) === index ? newModule : oldModule,
+        );
+      else newModules = [...state.course.modules, newModule];
```

Comment:
For legibility, could you change this to use braces? Just because the content of the if statement spans multiple lines.